test/verify/check-machines-dbus

```diff
@@ -522,13 +547,29 @@ class TestMachinesDBus(machineslib.TestMachines):
         b.wait_not_present("#vm-subVmTest1-disks-vde-device")
 
         # Test detaching disk of a paused domain
+        if args["logfile"] is not None:
+            m.execute("> {0}".format(args["logfile"])) # clear logfile
         m.execute("virsh start subVmTest1")
         # Make sure that the VM booted normally before attempting to suspend it
         if args["logfile"] is not None:
             wait(lambda: "Linux version" in m.execute("cat {0}".format(args["logfile"])), delay=3)
         m.execute("virsh suspend subVmTest1")
         b.wait_in_text("#vm-subVmTest1-state", "paused")
         b.wait_attr("#delete-subVmTest1-disk-vdf", "disabled", "")
+        m.execute("virsh resume subVmTest1")
+
+        # Test detaching of disk on non-persistent VM
+        m.execute("virsh undefine subVmTest1")
+        m.execute("virsh attach-disk --domain subVmTest1 --source /mnt/vm_one/mydiskofpoolone_1 --target vdd --targetbus virtio")
+        b.wait_present("#vm-subVmTest1-disks-vdd-device")
+        b.click("#delete-subVmTest1-disk-vdd")
+        b.wait_present(".modal-dialog")
+        b.click(".modal-footer button:contains(Remove)")
+        i = 0
+        while b.is_present(".modal-dialog") and i < 3:
```

Comment:
What is this?

Reply:
Oh this is a hacky solution of something I could not quite figured out. As we talked before, detaching disks may sometimes be refused by kernel on first try (even if kernel is booted, for example disk is used by kernel) and no error return would be returned.
My guess is that there is just too much what could go wrong when detaching stuff (there is a whole project, guest agent, which solves communication between guest and host to get some info about these issues. But we use cirros in our nested guest and cirros doesn't have guest agent).
Apparently sending detach action second time will work. (I see I do it 3 times just to be sure, but only twice should work). It's quick easy solution for now. But yeah, it's hacky.

Reply:
Are you sure it's not related to this bug we had? https://github.com/cockpit-project/cockpit/pull/13488

Reply:
If it's really just the dialog error, then we can test this without the retry loop. But in the [discussion above](https://github.com/cockpit-project/cockpit/pull/13241#issuecomment-572296750) it seemed like something actually keeps the disk busy, and it's totally credible that e. g. udev accesses the new device right after you add it. The above discussion suggests some better long-term approaches how to test this properly.

Reply:
I think it was really related to the issue with the dialog not cliosing and not the above assumption. I tried it locally rebased on the current master which contains the fix for the modals not closing, and it didn't even fail once.
I added a fix up commit removing these WAs, hopefully it will work now - if not feel free to remove again.
(I also rebased it - it had some merge conflicts)